common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -1417,6 +1417,13 @@ def save_problem(self, data):
             }
 
         self.lcp.student_answers = answers
+        answer_keys = [key for key in answers]
+        for answer_key in answer_keys:
+            # Reset the correctness property to default(None) state when attempted wrong
+            if self.lcp.correct_map.get_property(answer_key, 'correctness', None) == 'incorrect':
```

Comment:
Also, we get the same bug the other way around too: If you answer a problem correctly, then select another response, click save, then refresh the page, it looks like an incorrect answer is correct, like in the screenshot below [from your sandbox](http://problemsave.sandbox.edx.org/courses/course-v1:mushtaq+101+2015/courseware/482376738a92445493faded4e7ebdbd5/2bc99a20a2394dc58ebaa05cc35f8bf2/1?activate_block_id=block-v1%3Amushtaq%2B101%2B2015%2Btype%40vertical%2Bblock%404642d49266af43f6a21fee9d6c7c2c38).

So, we need some other logic to tell us whether or not we should display the check, rather than just looking at a an answer's correctness.




![aw_4vihi-ok-pu-fy3lnlj-mccygxsv5gh1uvgzpody](https://cloud.githubusercontent.com/assets/2748698/9743992/47cdd0d2-5637-11e5-98d0-e6e4ba6e529d.png)